### DISCUSSION TITLE
Toc - Nested vs Flat

### DISCUSSION THREAD

**Discussion by SharePointing** [vuejs/vitepress#2951] [Q&A]
I'm noticing that there's a couple of different views of the toc (on the right) - flat and nested. 

Anyone know how to force the nested view ? 

I'm sure I'm just not reading the docs right, but it's no where I could find.

on this site - there's both examples

![toc-flat](https://github.com/vuejs/vitepress/assets/5741429/0e1bcd32-0c8c-4625-a417-49101cadaf1b)
Flat TOC

![toc-nested](https://github.com/vuejs/vitepress/assets/5741429/473a5dbf-703a-428d-921a-623492aa9687)
Nested TOC

I'd like to have all pages be nested, but all of my sites just show flat down to second level and do not show the third level.

**Comment by SharePointing** [comment]
Nothing like discovering the answer after you post the question :-)

I was able to add the frontmatter for outline: deep to the header of the page to force what I was wanting for the toc. 

---
outline: deep
---

Hopefully it's helpful to some other poor soul !

**Reply by brc-dd** [reply]
You can also enable it site-wide by setting outline deep in your themeConfig.

**Reply by SharePointing** [reply]
Yes - That's helpful !

I'm thinking the option needs to be called out in the site config docs. I'll submit an update for the next 100 people that need it !
